job-server-extras/test/spark.jobserver/HiveJobSpec.scala

```diff
@@ -9,8 +10,15 @@ import spark.jobserver.context.{HiveContextLike, HiveContextFactory}
 import spark.jobserver.io.{JobDAO, JobDAOActor}
 
 class TestHiveContextFactory extends HiveContextFactory {
-  override protected def contextFactory(conf: SparkConf): C =
-    new TestHiveContext(new SparkContext(conf)) with HiveContextLike
+  override protected def contextFactory(conf: SparkConf): C = {
+    val sc = new SparkContext(conf)
+    Try(new TestHiveContext(sc) with HiveContextLike) match {
+      case Success(hc) => hc
+      case Failure(e) =>
+        sc.stop
+        throw e
+    }
+  }
```

Comment:
This was required as this test fails on my Windows machine and it did not clean-up after itself.